packages/terra-button/src/Button.jsx

```diff
@@ -139,7 +139,9 @@ class Button extends React.Component {
     // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
     // This will put focus on the button when clicked if it is not currently the active element.
     if (document.activeElement !== event.currentTarget) {
+      this.shouldShowFocus = false;
```

Comment:
do we need this..? because `this.shouldShowFocus` is being set to true in below line.

Reply:
Yes we need it. `this.shouldShowFocus` is set true after firing the focus event.